ion-c-sys/src/reader.rs

```diff
@@ -55,6 +55,317 @@ impl<'a> IonCReaderHandle<'a> {
+    ///
+    /// assert_eq!(ION_TYPE_STRUCT, reader.next()?);
+    /// assert!(reader.get_field_name()?.value.is_null());
+    /// reader.step_in()?;
+    /// assert!(reader.get_field_name()?.value.is_null());
+    /// assert_eq!(ION_TYPE_INT, reader.next()?);
+    /// assert_eq!("a", reader.get_field_name()?.try_as_str()?);
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[inline]
+    pub fn get_field_name(&mut self) -> IonCResult<IonCStringRef> {
+        let mut field = ION_STRING::default();
+        ionc!(ion_reader_get_field_name(self.reader, &mut field))?;
+
+        Ok(IonCStringRef::new(self, field))
+    }
+
+    // TODO annotation support
+
+    /// Reads a `bool` value from the reader.
+    ///
+    /// ## Usage
+    /// ```
+    /// # use std::convert::*;
+    /// # use ion_c_sys::*;
+    /// # use ion_c_sys::reader::*;
+    /// # use ion_c_sys::result::*;
+    /// # fn main() -> IonCResult<()> {
+    /// let mut buf = b"true".to_vec();
+    /// let mut reader = IonCReaderHandle::try_from(buf.as_mut_slice())?;
+    ///
+    /// assert_eq!(ION_TYPE_BOOL, reader.next()?);
+    /// assert!(reader.read_bool()?);
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[inline]
+    pub fn read_bool(&mut self) -> IonCResult<bool> {
+        let mut value = 0;
+        ionc!(ion_reader_read_bool(self.reader, &mut value))?;
+
+        Ok(value != 0)
+    }
+
+    /// Reads an `int` value from the reader.
+    ///
+    /// ## Usage
+    /// ```
+    /// # use std::convert::*;
+    /// # use ion_c_sys::*;
+    /// # use ion_c_sys::reader::*;
+    /// # use ion_c_sys::result::*;
+    /// # fn main() -> IonCResult<()> {
+    /// let mut buf = b"42".to_vec();
+    /// let mut reader = IonCReaderHandle::try_from(buf.as_mut_slice())?;
+    ///
+    /// assert_eq!(ION_TYPE_INT, reader.next()?);
+    /// assert_eq!(42, reader.read_i64()?);
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[inline]
+    pub fn read_i64(&mut self) -> IonCResult<i64> {
+        let mut value = 0;
+        ionc!(ion_reader_read_int64(self.reader, &mut value))?;
+
+        Ok(value)
+    }
+
+    /// Reads a `float` value from the reader.
+    ///
+    /// ## Usage
+    /// ```
+    /// # use std::convert::*;
+    /// # use ion_c_sys::*;
+    /// # use ion_c_sys::reader::*;
+    /// # use ion_c_sys::result::*;
+    /// # fn main() -> IonCResult<()> {
+    /// let mut buf = b"3.0e0".to_vec();
+    /// let mut reader = IonCReaderHandle::try_from(buf.as_mut_slice())?;
+    ///
+    /// assert_eq!(ION_TYPE_FLOAT, reader.next()?);
+    /// assert_eq!(3.0, reader.read_f64()?);
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[inline]
+    pub fn read_f64(&mut self) -> IonCResult<f64> {
+        let mut value = 0.0;
+        ionc!(ion_reader_read_double(self.reader, &mut value))?;
+
+        Ok(value)
+    }
+
+    // TODO support ION_INT (arbitrary large integer) reads
+    // TODO support decimal reads
+    // TODO support timestamp reads
```

Comment:
Not sure how we want to keep track of these TODOs as issues for the ion-c wrapper. Perhaps a tag in our repo's issues?

Reply:
I have some issues created--I will add the references in the code.